scala2-functional-program-design/week2/streams/src/main/scala/streams/Solver.scala

```diff
@@ -81,5 +90,8 @@ trait Solver extends GameDef {
    * the first move that the player should perform from the starting
    * position.
    */
-  lazy val solution: List[Move] = ???
+  lazy val solution: List[Move] = pathsToGoal match {
+    case LazyList() => List()
+    case (_: Block, moves: List[Move]) #:: _ => moves
```

Comment:
Is it the right order? 
Please recheck

Reply:
Do you mean the case order or the runtime code? I've tested it and it seems to work well using the build it tester

Reply:
OK, because I have a .reverse at the end of that line 😕

Reply:
been there.. :)